core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -115,12 +115,16 @@ public static GetField fromXContent(XContentParser parser) throws IOException {
         ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
         String fieldName = parser.currentName();
         XContentParser.Token token = parser.nextToken();
-        ensureExpectedToken(XContentParser.Token.START_ARRAY, token, parser::getTokenLocation);
-        List<Object> values = new ArrayList<>();
-        while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            values.add(parseStoredFieldsValue(parser));
+        if (token == XContentParser.Token.START_ARRAY) {
+            List<Object> values = new ArrayList<>();
+            while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                values.add(parseStoredFieldsValue(parser));
+            }
+            return new GetField(fieldName, values);
+        } else {
+            parser.skipChildren();
```

Comment:
I wonder if we are making this parser too lenient with this change. it's ok to skip what we don't know about, but maybe we can avoid returning null as the array with at least a value is required?